### ISSUE TITLE
mandoc incorrectly reports unknown macros in man page

### ISSUE THREAD

**Issue by jfriesse** [rpminspect/rpminspect#132]
Trying to run rpminspect (Fedora 31 version - rpminspect-0.4-1.fc31.x86_64) on omping package and man checker reports following problems for man page (https://github.com/jfriesse/omping/blob/master/omping.8):
```
Error parsing omping.8.gz:17:1: ERROR: skipping unknown macro: .Dd Jun 22, 2011
Error parsing omping.8.gz:18:1: ERROR: skipping unknown macro: .Dt OMPING 8
Error parsing omping.8.gz:19:1: ERROR: skipping unknown macro: .Os
Error parsing omping.8.gz:21:1: ERROR: skipping unknown macro: .Sh NAME
Error parsing omping.8.gz:22:1: ERROR: skipping unknown macro: .Nm omping
Error parsing omping.8.gz:23:1: ERROR: skipping unknown macro: .Nd test IP multicast
Error parsing omping.8.gz:26:1: ERROR: skipping unknown macro: .Op Fl 46CDEFqVv
Error parsing omping.8.gz:39:1: ERROR: skipping unknown macro: .Ar remote_addr...
Error parsing omping.8.gz:46:1: ERROR: skipping unknown macro: .Xr ping 8
Error parsing omping.8.gz:64:1: ERROR: skipping unknown macro: .Bl -tag -width Ds
Error parsing omping.8.gz:65:1: ERROR: skipping unknown macro: .It Fl 4
Error parsing omping.8.gz:104:1: ERROR: skipping unknown macro: .Cm asm
Error parsing omping.8.gz:143:1: ERROR: skipping unknown macro: .Fl i
Error parsing omping.8.gz:170:1: ERROR: skipping unknown macro: .El
Error parsing omping.8.gz:171:1: ERROR: skipping unknown macro: .Pp
Error parsing omping.8.gz:181:1: ERROR: skipping unknown macro: .Ex -std
Error parsing omping.8.gz:187:1: ERROR: skipping unknown macro: .Dl $ omping node-01 node-02 node-03
Error parsing omping.8.gz:257:1: ERROR: skipping unknown macro: .An Jan Friesse Aq jfriesse@redhat.com .
```

According to mdoc syntax (https://mandoc.bsd.lv/man/mdoc.7.html) all these macros are valid. Man page also displays correctly in both of mandoc man (as packaged in void linux - mdocm 1.14.5) and mandb (man-db-2.8.4-5.fc31.x86_64)

I haven't yet take a look to rpminspect source yet, but I believe there is some (hopefully small) fix needed to parse mdoc macros properly.

**Comment by dcantrell** [comment]
What is the NVR of the package that rpminspect is looking at?

**Comment by dcantrell** [comment]
The validation is using MPARSE_MAN|MPARSE_UTF8|MPARSE_LATIN1.  To better handle -man vs. -mdoc validation, I think maybe the inspection needs two passes.  First try what is currently done.  On failure, try again for mdoc.  Unless mparse_alloc() will allow me to do MPARSE_MAN|MPARSE_MDOC but that seems like it won't work.  I will look in to this one after the issue I am currently investigating.

**Comment by jfriesse** [comment]
@dcantrell Thank you for the info. Yep, solution sounds good.

You may also consider using different one. According to doc - http://mandoc.bsd.lv/man/mandoc.3.html - `mparse_alloc`:
> When the MPARSE_MDOC or MPARSE_MAN bit is set, only that parser is used. Otherwise, the document type is automatically detected.

So when nether one is used it should autodetect document type. IMHO this is what man uses internally so user hasn't had to specify `-man` or `-mdoc` argument and (again IMHO) it may be best for rpminspect because it behaves as user would.

**Comment by dcantrell** [comment]
Looking back at your original comment, I see that you are using an older version of rpminspect (rpminspect-0.4-1.fc31.x86_64).  I have tried to reproduce your issue with the latest in the master branch and am unable.  Here's what I get:

```
[dcantrell@awvr rpminspect]$ fed-rpminspect -T manpage omping-0.0.4-21.fc33
man-pages:
----------
Result: OK
Waiver Authorization: Not Waivable
```

It looks like things are working with the latest release.  Can you confirm on your end using rpminspect from the Copr repo?  https://copr.fedorainfracloud.org/coprs/dcantrell/rpminspect/

**Comment by dcantrell** [comment]
Nevermind, I believe I have found a reproducer.  Working on a fix.

**Comment by dcantrell** [comment]
Looks like removing the MPARSE_MAN define did the trick.  The type is now correctly detected:

```
$ fed-rpminspect -T manpage omping-0.0.4-21.fc33
man-pages:
----------
Result: OK
Waiver Authorization: Not Waivable
```

I also tested it on the calendar package which I maintain.  calendar is from OpenBSD and uses mdoc and it passes with the change.

**Comment by jfriesse** [comment]
@dcantrell Haven't tested copr yet, but it looks like it is no longer needed. Thank you for the fix!
